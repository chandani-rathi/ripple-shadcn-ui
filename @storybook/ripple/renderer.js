// renderer.js
export async function renderToCanvas({ storyFn, canvasElement }) {
  console.log("renderer")
  const output = storyFn();

  // Example: simple HTML render
  if (typeof output === 'string') {
    canvasElement.innerHTML = output;
  } else if (output instanceof HTMLElement) {
    canvasElement.innerHTML = '';
    canvasElement.appendChild(output);
  } else {
    canvasElement.innerHTML = `<pre>Unsupported output: ${JSON.stringify(output)}</pre>`;
  }
}
