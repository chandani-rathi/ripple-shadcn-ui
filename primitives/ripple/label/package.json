{
  "name": "@ripple-primitives/label",
  "version": "1.0.0",
  "type": "module",
  "source": "./src/index.ts",
  "main": "./src/index.ts",
  "module": "./src/index.ts",
  "exports": {
    ".": {
      "import": "./dist/index.es.js",
      "require": "./dist/index.cjs.js"
    }
  },
  "scripts": {
    "build": "vite build",
    "watch": "vite build --watch",
    "test": "vitest run",
    "test:watch": "vitest"
  },
  "dependencies": {
    "@ripple-primitives/primitive-ui": "workspace:*",
    "ripple": "catalog:"
  },
  "packageManager": "pnpm@10.20.0"
}
