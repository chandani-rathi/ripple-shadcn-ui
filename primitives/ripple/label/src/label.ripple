import { Primitive } from '@ripple-primitives/primitive-ui';
const NAME = 'LABEL';

export component Label(props) {
  <Primitive.label //{...props}
    for={props.htmlFor}
    class={props.class}
    onMouseDown={(event) => {
        // only prevent text selection if clicking inside the label itself
        const target = event.target as HTMLElement;
        if (target.closest('button, input, select, textarea')) return;

        props.onMouseDown?.(event);
        // prevent text selection when double clicking label
        if (!event.defaultPrevented && event.detail > 1) event.preventDefault();
    }}
    >
    <props.@children />
  </Primitive.label>
}