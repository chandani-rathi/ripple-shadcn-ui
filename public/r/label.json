{
  "$schema": "https://ui.shadcn.com/schema/registry-item.json",
  "name": "label",
  "type": "registry:ui",
  "dependencies": [],
  "devDependencies": [],
  "registryDependencies": [],
  "files": [
    {
      "path": "registry/new-york/ui/label.ripple",
      "content": "import type { Component } from 'ripple';\r\nimport { cn } from \"@/lib/utils\"\r\n\r\nexport component Label(props: { $children: Component, $class: string, $htmlFor: string }) {\r\n\t<label\r\n        data-slot=\"label\"\r\n        $for={props.$htmlFor}\r\n        $class={cn(\r\n            \"flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50\", \r\n            props.$class\r\n        )}\r\n        onMouseDown={(event) => {\r\n            // only prevent text selection if clicking inside the label itself\r\n            const target = event.target;\r\n            if (target.closest('button, input, select, textarea')) return;\r\n\r\n            props.onMouseDown?.(event);\r\n            // prevent text selection when double clicking label\r\n            if (!event.defaultPrevented && event.detail > 1) event.preventDefault();\r\n        }}\r\n      >\r\n        if(props.$children) {\r\n            <props.$children />\r\n        }\r\n    </label>\r\n}\r\n",
      "type": "registry:ui"
    }
  ]
}