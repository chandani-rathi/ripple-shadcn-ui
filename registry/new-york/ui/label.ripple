import type { Component } from 'ripple';
import { cn } from "@/lib/utils"

export component Label(props: { $children: Component, $class: string, $htmlFor: string }) {
	<label
        data-slot="label"
        $for ={props.$htmlFor}
        $class={cn(
            "flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50", 
            props.$class
        )}
        onMouseDown={(event) => {
            console.log(event);
            // only prevent text selection if clicking inside the label itself
            const target = event.target;
            if (target.closest('button, input, select, textarea')) return;

            props.onMouseDown?.(event);
            // prevent text selection when double clicking label
            if (!event.defaultPrevented && event.detail > 1) event.preventDefault();
        }}
      >
        if(props.$children) {
            <props.$children />
        }
    </label>
}
