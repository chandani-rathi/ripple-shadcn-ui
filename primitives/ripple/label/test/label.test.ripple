import { axe } from 'vitest-axe';
import { Label }  from "../src/label.ripple";

const TEXT_CHILD = "TEST"

describe("label", () => {
    beforeEach(() => {
		component WrappingControl(){
			<Label>
				{TEXT_CHILD} <input type="number" />
			</Label>
		}
		component ReferencingControl(){
			<Label htmlFor="control">{TEXT_CHILD}</Label>
			<button id="control">{"Control"}</button>
		}
		component Test(){
			<WrappingControl />
			<ReferencingControl />
		}
		render(Test);
	});

    it('should have no accessibility violations', async () => {
		expect(await axe(container)).toHaveNoViolations();
	});
    
    it("should work", () => {
         expect(container).toMatchSnapshot()
    })
})