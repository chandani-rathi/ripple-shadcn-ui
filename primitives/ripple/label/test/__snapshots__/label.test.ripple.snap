// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`label > should work 1`] = `
<div>
  <!---->
  <!---->
  <!---->
  <!---->
  <label>
    <!---->
    <!---->
    <!---->
    TEST
    <input
      type="number"
    />
    <!---->
    <!---->
    <!---->
    <!---->
  </label>
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <label
    for="control"
  >
    <!---->
    <!---->
    <!---->
    TEST
    <!---->
    <!---->
    <!---->
    <!---->
  </label>
  <!---->
  <!---->
  <!---->
  <button
    id="control"
  >
    Control
  </button>
  <!---->
  
</div>
`;
